File: discord_bot/slash_commands_setup.py
import logging

import discord

from discord_bot.slash_commands.hello_world import add_hello_world_slash_command
from discord_bot.slash_commands.check_repos import add_check_repos_slash_command
from discord_bot.discord_bot import SggBot

log = logging.getLogger(__name__)


def add_all_slash_commands(bot: SggBot, server_id):
    guild = discord.Object(id=int(server_id))
    add_hello_world_slash_command(bot, guild)
    add_check_repos_slash_command(bot, guild)

    @bot.event
    async def on_ready():  # pyright: ignore[reportUnusedFunction]
        await bot.tree.sync(guild=guild)
        log.info("Slash commands synced")

File: discord_bot/slash_commands_setup.go
package discord_bot

import (
	"log"

	"github.com/bwmarrin/discordgo"

	"sgg-discord-bot/discord_bot/slash_commands"
)

func AddAllSlashCommands(session *discordgo.Session, guildID string) {
	commands := []*discordgo.ApplicationCommand{
		slash_commands.HelloWorldCommand,
		slash_commands.CheckReposCommand,
	}

	session.AddHandler(func(s *discordgo.Session, i *discordgo.InteractionCreate) {
		if i.Type != discordgo.InteractionApplicationCommand {
			return
		}
		switch i.ApplicationCommandData().Name {
		case "hello-world":
			slash_commands.HandleHelloWorld(s, i)
		case "check-repos":
			slash_commands.HandleCheckRepos(s, i)
		}
	})

	session.AddHandler(func(s *discordgo.Session, r *discordgo.Ready) {
		for _, cmd := range commands {
			_, err := s.ApplicationCommandCreate(s.State.User.ID, guildID, cmd)
			if err != nil {
				log.Printf("Error creating command %s: %v", cmd.Name, err)
			}
		}
		log.Println("Slash commands synced")
	})
}
